src/core/mac/mac.hpp

```diff
@@ -820,6 +818,11 @@ class Mac : public InstanceLocator, private NonCopyable
 #if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
     TimeMilli mCslTxFireTime;
 #endif
+#endif
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    // When Mac::mCslChannel is 0, it indicates that CSL channel has not been specified by the upper layer.
+    uint8_t  mCslChannel;
+    uint16_t mCslPeriod;
```

Comment:
This is intentional, since it's only 3B but saves some handling code. `CSL Period = 0` is an indication for the SubMac that CSL sampling is active, while we still want to keep the user configured value somewhere.

Reply:
Can we add a bool in `SubMac` to directly indicate this (i.e., whether  CSL sampling is active or not) instead of using period zero as special value for this purpose.

I feel having same variables for `mCslChannel` and `mCslPeriod` in both `Mac` and `SubMac` is not a good idea. (will be confusing for someone reading the code).

Reply:
At a first glance doing so will increase code inefficiency, missing the opportunity to reuse `Mac::UpdateCsl()`  for verification that CSL is supported at the moment or scheduling Child Update Request. Also it will increase interaction with SubMac (serialization) for checking those parameters.